packages/jest-cli/src/TestSequencer.js

```diff
@@ -59,6 +71,37 @@ export default class TestSequencer {
     return cache;
   }
 
+  _fileSizeRecurseDependencies(
+    test: Test,
+    stats: {[path: Path]: number},
+  ): number {
+    const {
+      path,
+      context: {resolver, hasteFS, config},
+    } = test;
+
+    const fileSize = path =>
+      stats[path] ||
+      (resolver.isCoreModule(path)
+        ? coreModuleWeights[path] || 0
+        : (stats[path] = fs.statSync(path).size));
+
+    const dependencyResolver = new DependencyResolver(
+      resolver,
+      hasteFS,
+      buildSnapshotResolver(config),
+    );
+    const recursiveDependencies = dependencyResolver.resolveRecursive(path, {
+      includeCoreModules: true,
+    });
+    const size =
+      Array.from(recursiveDependencies)
+        .map(file => fileSize(file))
+        .reduce((sum, size) => sum + size, 0) + fileSize(path);
```

Comment:
```suggestion
        .reduce((sum, file) => sum + fileSize(file), 0) + fileSize(path);
```

This will save a full loop - we wanna be as quick as possible here 🙂